README.md

```diff
@@ -362,29 +368,61 @@ Benchee.run %{
    after_each: fn(result) -> assert result == 42 end
 ```
 
-When might this be useful?
-* You can use this to run assertions that whatever your benchmarking function returns is truly what it should be (i.e. all "contestants" work as expected)
+_When might this be useful?_
+
+* Running assertions that whatever your benchmarking function returns is truly what it should be (i.e. all "contestants" work as expected)
 * Busting caches/terminating processes setup in `before_each` or elsewhere
 * Deleting files created by the benchmarking function
 
-### after_scenario
+#### Hook arguments and return values
+
+Before hooks form a chain, where the return value of the previous hook becomes the argument for the next one. The first defined `before` hook receives the scenario input as an argument, and returns a value that becomes the argument of the next in the chain. The benchmarking function receives the value of the last `before` hook as its argument (or the scenario input if there are no `before` hooks).
 
-Is executed after a scenario completed and is passed whatever the return value of the last `before_scenario` was, if there was none it is passed the specified `input`. Otherwise it works much like [`after_each`](#after_each).
+After hooks do not form a chain, and their return values are discarded. An `after_each` hook receives the return value of the benchmarking function as its argument. An `after_scenario` function receives the return value of its `before_scenario` counterpart (or the scenario's input if there is no `before_scenario` hook).
```

Comment:
Slightly incorrect here (and also in the `before_scenario`) part - the `after_scenario` receives the output of the last before_scenario as an input. Aka if there is only a global one both global and local receives its output, if there is a local one both receive that. 

We don't wanna enforce that there always needs to be a counterpart, so they just get the return value of the before_scenario the furthest down the chain